Rover/AP_MotorsUGV.cpp

```diff
@@ -613,9 +615,52 @@ void AP_MotorsUGV::output_regular(bool armed, float ground_speed, float steering
         if (_scale_steering) {
             // vectored thrust handling
             if (have_vectored_thrust()) {
-                if (fabsf(throttle) > _vector_throttle_base) {
-                    // scale steering down linearly as throttle increases above _vector_throttle_base
-                    steering *= constrain_float(_vector_throttle_base / fabsf(throttle), 0.0f, 1.0f);
+
+                // normalise desired steering and throttle to ease calculations
+                float steering_norm = steering / 4500.0f;
+                const float throttle_norm = throttle / 1000.0f;
+
+                // steering can never be more than throttle
+                if (fabsf(steering_norm) > fabsf(throttle_norm)) {
+                    if (is_positive(steering_norm)) {
+                        steering_norm = fabsf(throttle_norm);
+                        limit.steer_right = true;
+                    }
+                    if (is_negative(steering_norm)) {
+                        steering_norm = -fabsf(throttle_norm);
+                        limit.steer_left = true;
+                    }
+                }
+
+                if (!is_zero(throttle_norm)) {
+                    // calculate steering angle
+                    float steering_angle_rad = tanf(steering_norm / throttle_norm);
+                    const float vector_angle_max_rad = radians(constrain_float(_vector_angle_max, 0.0f, 90.0f));
+                    // limit steering angle to vector_angle_max
+                    if (fabsf(steering_angle_rad) > vector_angle_max_rad) {
+                        steering_angle_rad = constrain_float(steering_angle_rad, -vector_angle_max_rad, vector_angle_max_rad);
+                        if (is_positive(steering_norm)) {
+                            limit.steer_right = true;
+                        }
+                        if (is_negative(steering_norm)) {
+                            limit.steer_left = true;
+                        }
+                     }
+
+                    // convert steering angle to steering output
+                    steering = steering_angle_rad / vector_angle_max_rad * 4500.0f;
+
+                    // scale up throttle to compensate for steering angle
+                    const float throttle_scaler_inv = cosf(steering_angle_rad);
+                    if (!is_zero(throttle_scaler_inv)) {
+                        throttle /= throttle_scaler_inv;
+                        if (throttle >= 100.0f) {
+                            limit.throttle_upper = true;
+                        }
+                        if (throttle <= -100.0f) {
+                            limit.throttle_lower = true;
+                        }
+                    }
```

Comment:
Scaling the throttle up to preserve the longitudinal component of thrust will scale up the lateral component as well - which will amplify the yaw effect as the steering angle is increased.